### PR TITLE
doc/rbd/rbd-kubernetes: add ceph.conf configmap

### PR DESCRIPTION
**Pull request by Madhu-1**
Fix for https://github.com/ceph/ceph-csi/issues/2535

CC @idryomov

### CODE REVIEW

File: doc/rbd/rbd-kubernetes.rst [hunk 1]
```diff
@@ -134,6 +134,39 @@ Once generated, store the new `ConfigMap` object in Kubernetes::
 
         $ kubectl apply -f csi-kms-config-map.yaml
 
+Recent versions of `ceph-csi` also require an additional `ConfigMap` object
+to define Ceph configuration to add to ceph.conf file inside CSI containers.
```
Comment (idryomov) [inline doc/rbd/rbd-kubernetes.rst:138]:
```suggestion
to define Ceph configuration to add to ceph.conf file inside CSI containers::
```

File: doc/rbd/rbd-kubernetes.rst [hunk 2]
```diff
@@ -134,6 +134,39 @@ Once generated, store the new `ConfigMap` object in Kubernetes::
...
+refer sample ceph.conf
+at https://github.com/ceph/ceph-csi/tree/master/examples::
```
Comment (idryomov) [inline doc/rbd/rbd-kubernetes.rst:140]:
```suggestion
```

File: doc/rbd/rbd-kubernetes.rst [hunk 3]
```diff
@@ -134,6 +134,39 @@ Once generated, store the new `ConfigMap` object in Kubernetes::
...
+            auth_service_required = cephx
+            auth_client_required = cephx
+
+            # Workaround for http://tracker.ceph.com/issues/23446
+            fuse_set_user_groups = false
+
+            # ceph-fuse which uses libfuse2 by default has write buffer size of 2KiB
+            # adding 'fuse_big_writes = true' option by default to override this limit
+            # see https://github.com/ceph/ceph-csi/issues/1928
+            fuse_big_writes = true
```
Comment (idryomov) [inline doc/rbd/rbd-kubernetes.rst:159]: This is an rbd guide so let's drop ceph-fuse specfic stuff.

**Review by idryomov — CHANGES_REQUESTED** [review]

File: doc/rbd/rbd-kubernetes.rst [hunk 4]
```diff
@@ -134,6 +134,29 @@ Once generated, store the new `ConfigMap` object in Kubernetes::
...
+        apiVersion: v1
+        kind: ConfigMap
+        data:
+          ceph.conf: |
+            [global]
+            auth_cluster_required = cephx
+            auth_service_required = cephx
+            auth_client_required = cephx
+            # keyring is a required key and its value should be empty
+          keyring: |
```
Comment (idryomov) [inline doc/rbd/rbd-kubernetes.rst:151]: keyring comment is misindented

File: doc/rbd/rbd-kubernetes.rst [hunk 5]
```diff
@@ -134,6 +134,29 @@ Once generated, store the new `ConfigMap` object in Kubernetes::
 
         $ kubectl apply -f csi-kms-config-map.yaml
 
+Recent versions of `ceph-csi` also require an additional `ConfigMap` object
+to define Ceph configuration to add to ceph.conf file inside CephCSI containers::
+
+        $ cat <<EOF > ceph-conf.yaml
```
Comment (idryomov) [inline doc/rbd/rbd-kubernetes.rst:140]: Nit: let's match the other ConfigMap snippets:
```suggestion
        $ cat <<EOF > ceph-config-map.yaml
```

File: doc/rbd/rbd-kubernetes.rst [hunk 6]
```diff
@@ -134,6 +134,29 @@ Once generated, store the new `ConfigMap` object in Kubernetes::
...
+            auth_client_required = cephx
+            # keyring is a required key and its value should be empty
+          keyring: |
+        metadata:
+          name: ceph-config
+        EOF
+
+Once generated, store the new `ConfigMap` object in Kubernetes::
+
+        $ kubectl apply -f ceph-conf.yaml
```
Comment (idryomov) [inline doc/rbd/rbd-kubernetes.rst:158]:
```suggestion
        $ kubectl apply -f ceph-config-map.yaml
```

File: doc/rbd/rbd-kubernetes.rst (hunk 4 above)
Comment (Madhu-1) [reply doc/rbd/rbd-kubernetes.rst:151]: Fixed now

**Review by idryomov — APPROVED** [review]

**Pull request merged**